kedro-datasets/kedro_datasets/ibis/file_dataset.py

```diff
@@ -157,7 +157,7 @@ def hashable(value):
 
             config = deepcopy(self._connection_config)
             backend_attr = config.pop("backend") if config else None
-            backend = getattr(ibis, backend_attr)
+            backend = getattr(ibis, str(backend_attr))
```

Comment:
@ankatiyar this doesn't make sense.

The mypy error was valid. Now you're coercing `None` to a string just to make mypy happy, but the `getattr` will fail given `"None"`.

I was trying to figure out why the logic is like this to begin with, and the issue originates from the change in https://github.com/kedro-org/kedro-plugins/commit/c30434ce2a4485a64fc3f364f1de1c4f986a1ab5.

Reply:
oh my bad, would it make sense to do the following? :
```
if backend_attr:
  backend = getattr(ibis, backend_attr)
```

Reply:
The reason there was no default is because it's coded such that providing a backend was required. Let me see about providing a default backend; I'll raise a PR and ask for your review!

Reply:
Yeah makes sense, I was focussed on appeasing `mypy` didn't dig too deep into why it was written like this. Thank you! :)